Excercise_2.js

```diff
@@ -1 +1,67 @@
-//Implement Stack using Linked List.
\ No newline at end of file
+//Implement Stack using Linked List.
+
+class Node {
+    constructor(val){
+        this.val = val;
+        this.next = null;
+    }
+}
+
+class SinglyLinkedList{
+    constructor(){
+        this.head = null;
+        this.tail = null;
```

Comment:
Can we do this with just head pointer. Here is an algorithm for you
Push the node the head (that will be last in)
While popping.  Pop it from head itself. 

This way both Push and Pop functions will be O(1) time. However in your case the pop operation is O(n).

To get a better idea, see the PR by Raju (VenkateswaraRajuDatla)

Both insert and

Reply:
Thanks for the suggestions. If you look at Raju's code and if you push let's say 1-2-3 to his LinkedList you will see that linklist becomes this "3->2->1->null" but it should be the opposite "1->2->3->null" so I think it's a hack. Can you please take a look?

Reply:
What makes you think that its hack? 

If I am correct then you are thinking that LinkedList follows of an order of insertion. A new node has to be added to the tail always?

Reply:
It doesn't always have to be added to the end, a new node can be added anywhere but I think push means adding to the tail.